pkg/virt-controller/services/template_test.go

```diff
@@ -1662,9 +1711,13 @@ var _ = Describe("Template", func() {
 
 				pod, err := svc.RenderLaunchManifest(&vmi)
 				Expect(err).ToNot(HaveOccurred())
-
-				Expect(pod.Spec.Containers[0].Resources.Requests.Memory().ToDec().ScaledValue(resource.Mega)).To(Equal(int64(179)))
-				Expect(pod.Spec.Containers[0].Resources.Limits.Memory().ToDec().ScaledValue(resource.Mega)).To(Equal(int64(179)))
+				if svc.IsARM64() {
```

Comment:
The whole template service has a lot of unit tests which only execute `arm64` scenarios when run on `arm64`. Let's tackle rearranging  these if/else blocks into table tests. We should not have the need for `arm` architecture to check these code flows in the controller.

It should be good enough to simply override the architecture table entry with the arch from the entry in the template creator.

Reply:
Hi Roman,
I am not sure if my understanding is right, take this test case as an example, I write a sample code.
```
It("should calculate vcpus overhead based on guest toplogy", func(arch string) {
...
expectedMemory := make(map[string][]int)
expectedMemory["arm64"] = []int{179,179}
expectedMemory["arm64"] = []int{313,313}
Expect(pod.Spec.Containers[0].Resources.Requests.Memory().ToDec().ScaledValue(resource.Mega)).To(Equal(int64(expectedMemory[arch][0])))
Expect(pod.Spec.Containers[0].Resources.Requests.Memory().ToDec().ScaledValue(resource.Mega)).To(Equal(int64(expectedMemory[arch][1])))
...},
table.Entry(svc.GetCpuArch()),
)
```

Reply:
Something like this:

```go
			table.Describe("should add cpu and memory constraints to a template", func(arch string, limit, request string) {
                                [...]
                                svc.ARCH= arch # !!!!!!!!!!!!!!!!!!!!!!!!
				pod, err := svc.RenderLaunchManifest(&vmi)
				Expect(err).ToNot(HaveOccurred())

                                [...]
				Expect(pod.Spec.Containers[0].Resources.Requests.Memory().String()).To(Equal(request))
				Expect(pod.Spec.Containers[0].Resources.Limits.Memory().String()).To(Equal(limit))
		         }, 
                         table.Entry("arm64", limit1, request1),
                         table.Entry("amd64", limit2, request2),
                      )}


```

Reply:
Hi Roman,
The pr has updated. There are some multi-arch tests that related to OVMFPath changed to table driven test, as the detection for OVMPath is done in virt-config
https://github.com/kubevirt/kubevirt/blob/008629479ab38edfd57864db3f6a2be9ac86c13d/pkg/virt-controller/services/template.go#L932

I am not sure if it is better to move OVMPath related code to pkg/virt-controller/services/template.go

Reply:
you can probably create the service and the clusterConfig inside the table and not inside the before-each. A common setup function should work, and we anyway have to set the arch already. Instead of that creating the service should work too.

Reply:
> and

Ok, let me give it a try.